### ISSUE TITLE
Permission issue with viewHive

### ISSUE THREAD

**Issue by sujeshchirackkal** [CommBank/maestro#430]
Throws permission issues intermittently while trying to write into multiple partitions using viewHive API.

This issue was reported during CDH 5 Upgrade. Observation was that cascading-hive was overriding the hive.metastore.execute.setugi and hence the jobs are trying to write into partitions as 'hive' user rather than the user who actually runs the job. Had tried setting the hive conf path explicitly via ops-logical environment script. But seems this hasn't really fixed the issue.
export HADOOP_CLASSPATH="$HADOOP_CLASSPATH:/etc/hive/conf"

Workaround: Written using view API and calls msck repair on table to update the metadata.

**Comment by stephanh** [comment]
I thought this was resolved as part of the CDH5 upgrade? What changed so that is now an issue again?

**Comment by sujeshchirackkal** [comment]
The fix by adding hive conf in the classpath was tested in cdh5.2.4. We had
one more last minute upgrade, due to other issues, and I assume we did not
test this case specifically after that. But this occurred for a few
projects where there are multiple partitions being written in a single run.


On Tuesday, 21 July 2015, Stephan Hoermann <notifications@github.com> wrote:

> I thought this was resolved as part of the CDH5 upgrade? What changed so
> that is now an issue again?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CommBank/maestro/issues/430#issuecomment-123127243>.
>


-- 
Regards,
Sujesh

**Comment by stephanh** [comment]
@AnnTomie I believe your squad encountered this a while ago and you were going to do some investigating?

**Comment by AnnTomie** [comment]
Had faced this issue while writing data into multiple partitions. But the code is remediated with controller now. In relation with this issue we found below:

https://cwiki.apache.org/confluence/display/Hive/Permission+Inheritance+in+Hive
https://issues.apache.org/jira/browse/HIVE-6648
https://issues.apache.org/jira/browse/HIVE-6892
https://issues.cloudera.org/browse/KITE-596
https://issues.cloudera.org/browse/KITE-615

**Comment by stephanh** [comment]
@AnnTomie are you saying this is a known issue with Hive that is fixed in later versions of Hive? What behaviour on our side triggers this bug?

**Comment by AnnTomie** [comment]
Yes it seems the issue has affected hive version .13.0 which we are currently using.
We write data into multiple partition in the  scenario of multiple file load.

**Comment by rowandavies** [comment]
@AnnTomie Does it seem reasonable to use the workaround Suresh suggested until we upgrade to CDH5.4+ which includes the fixes you mentioned?  Is this affecting more than a few projects?

**Comment by AnnTomie** [comment]
@rowandavies : Yes currently we are using workarounds to get over it. Writing into multiple partition occurs in mainly in catch-up loads

**Comment by jonjuscorpion666** [comment]
Raised ticket with cloudera :Case#: 71496

**Comment by rowandavies** [comment]
@AnnTomie Cloudera would like to reproduce this issue, and we'd like to know exactly where we stand.  What code do we have that causes it (when we remove workarounds)?

**Comment by AnnTomie** [comment]
@rowandavies : Sure will work on this, and provide sample code to reproduce the issue.

**Comment by shmookey** [comment]
We have observed two main behaviours contributing to this bug:
1. The submitter (via cascading-hive) informs the metastore a new partition before it has been written.
2. The metastore does not impersonate the submitter in calls to `add_partition`.
A workaround for (1) has been merged into ebenezer (https://github.com/CommBank/ebenezer/pull/130) and needs to be tested for a range of maestro jobs before it can go into production. (2) can be fixed by forcing the `SetUGI` option, but this approach will be made obsolete by the transition to kerberos.
